kiwi/storage/subformat/vmdk.py

```diff
@@ -168,3 +179,24 @@ def _create_vmware_settings_file(self):                 # noqa: C901
             raise KiwiTemplateError(
                 '%s: %s' % (type(e).__name__, format(e))
             )
+
+    def _inject_pvscsi_adapter_type(self):
+        """
+        QEMU does not support the pvscsi adapter type.
+        According to suggestions VMware giving to their customers, just
+        open the VMDK file, and change ddb.adapterType directly.
+        """
+        vmdk_image_name = self.get_target_file_path_for_format('vmdk')
+        vmdk_descriptor = None
+        with open(vmdk_image_name, 'rb') as vmdk:
+            vmdk.seek(512, 0)
+            vmdk_descriptor = bytes(vmdk.read(1024))
+
+        if vmdk_descriptor:
+            vmdk_descriptor = vmdk_descriptor.replace(
+                b'"lsilogic"', b'"pvscsi"'
+            )
+            with open(vmdk_image_name, 'r+b') as vmdk:
+                vmdk.seek(512, 0)
+                vmdk.write(vmdk_descriptor)
+                vmdk.seek(0, 2)
```

Comment:
Is this line needed? I don't see why the fd should be moved up to the end after writing the descriptor

Reply:
If you open with r+b meaning open for read and write and you close the file descriptor the file gets truncated at the position of the file pointer. Thus to my knowledge from the university it's required to position to the end before closing... might be old fashioned though and I'm open to change it if that's no longer needed.